velox/substrait/SubstraitToVeloxPlanValidator.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/substrait/SubstraitToVeloxPlan.h"
+
+namespace facebook::velox::substrait {
+
+/// This class is used to validate whether the computing of
+/// a Substrait plan is supported in Velox.
+class SubstraitToVeloxPlanValidator {
+ public:
+  /// Used to validate whether the computing of this type is supported.
+  bool validate(const ::substrait::Type& sType);
+
+  /// Used to validate whether the computing of this Aggregation is supported.
+  bool validate(const ::substrait::AggregateRel& sAgg);
+
+  /// Used to validate whether the computing of this Project is supported.
+  bool validate(const ::substrait::ProjectRel& sProject);
+
+  /// Used to validate whether the computing of this Filter is supported.
+  bool validate(const ::substrait::FilterRel& sFilter);
+
+  /// Used to validate whether the computing of this Read is supported.
+  bool validate(const ::substrait::ReadRel& sRead);
+
+  /// Used to validate whether the computing of this Rel is supported.
+  bool validate(const ::substrait::Rel& sRel);
+
+  /// Used to validate whether the computing of this RelRoot is supported.
+  bool validate(const ::substrait::RelRoot& sRoot);
+
+  /// Used to validate whether the computing of this Plan is supported.
+  bool validate(const ::substrait::Plan& sPlan);
+
+ private:
+  /// A query context used for function validation.
+  std::shared_ptr<core::QueryCtx> queryCtx_{core::QueryCtx::createForTest()};
+
+  /// A memory pool used for function validation.
+  std::unique_ptr<memory::MemoryPool> pool_{
+      memory::getDefaultScopedMemoryPool()};
+
+  /// An execution context used for function validation.
```

Comment:
Curious why do we need execution context.

Reply:
Looks like it's used to construct `ExprSet`.  And this is the only thing need to be pass in by constructor (no longer need `MemoryPool` or `QueryCtx`).

Reply:
Removed the memory pool and query context, and passed the execution context through constructor.